rotkehlchen/db/upgrades/v41_v42.py

```diff
@@ -198,6 +198,21 @@ def _remove_balancer_events_table(write_cursor: 'DBCursor') -> None:
     )
 
 
+def _delete_orphan_events(write_cursor: 'DBCursor') -> None:
+    """
+    Delete all the evm events that have a tx_hash that is not present in the db. This can happen
+    for customized events of addresses that got deleted from the database for example.
+
+    This query assumes that an evm event always has a evm_events_info but this is not the case
+    for zksync lite. Since prior to 1.33 that assumption holds this query is safe.
```

Comment:
?? Why do you make this assumption? Did you check the code? How can en `EvmEvent` not have an entry in evm_events_info. That's impossible.

```suggestion
```

Reply:
my fault here. I copied the wrong table. It is `evm_transactions` since I join with it. FOr zksync we have `zksynclite_transactions`